lib/rules/no-deprecated-router-transition-methods.js

```diff
@@ -0,0 +1,201 @@
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'enforce usage of router service transition methods',
+      category: 'Deprecations',
+      recommended: false,
+      url: 'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-deprecated-router-transition-methods.md',
+    },
+    fixable: 'code',
+    schema: [],
+    messages: {
+      main: 'Calling "{{methodUsed}}" in {{moduleType}}s is deprecated, call {{desiredMethod}} on the injected router service instead.',
+    },
+  },
+
+  create(context) {
+    // State being tracked for this file.
+    let serviceInjectImportName = undefined;
+    let routerServicePropertyName = undefined;
+    let isValidModule = false;
+
+    // State being tracked for the current class we're inside.
+    const classStack = new Stack();
+
+    // This rule does not apply to test files or non class modules
+    if (emberUtils.isTestFile(context.getFilename())) {
+      return {};
+    }
+
+    return {
+      ImportDeclaration(node) {
+        if (node.source.value === '@ember/service') {
+          serviceInjectImportName =
+            serviceInjectImportName || getImportIdentifier(node, '@ember/service', 'inject');
+        }
+      },
+
+      ClassDeclaration(node) {
+        if (emberUtils.isAnyEmberCoreModule(context, node)) {
+          for (const classMember of node.body.body) {
+            if (
+              decoratorUtils.isClassPropertyOrPropertyDefinitionWithDecorator(
+                classMember,
+                serviceInjectImportName
+              )
+            ) {
+              const decorator = decoratorUtils.getDecorator(classMember, serviceInjectImportName);
+
+              if (
+                types.isCallExpression(decorator.expression) &&
+                decorator.expression.arguments[0]
+              ) {
+                if (decorator.expression.arguments[0].value === 'router') {
+                  routerServicePropertyName = classMember.key.name;
+                }
+              } else if (classMember.key.name === 'router') {
+                routerServicePropertyName = classMember.key.name;
+              }
+            }
+          }
+          const isRoute = emberUtils.isEmberRoute(context, node);
+          const isController = emberUtils.isEmberController(context, node);
+          isValidModule = isRoute || isController;
+
+          classStack.push({
+            node,
+            serviceInjectImportName,
+            routerServicePropertyName,
+            isValidModule,
+            isRoute,
+            isController,
+          });
+        }
+      },
+
+      'ClassDeclaration:exit'(node) {
+        // Leaving current (native) class.
+        if (classStack.size() > 0 && classStack.peek().node === node) {
+          classStack.pop();
+        }
+      },
+
+      MemberExpression(node) {
+        if (!isValidModule) {
```

Comment:
According to test coverage checks, this line is not tested. Can you test it? And also be sure to monitor the test coverage to ensure every line in the new rule is covered.

```
yarn test --coverage lib/rules/no-deprecated-router-transition-methods.js
```